### ISSUE TITLE
Browser Launcher fails in Kodi 20.0

### ISSUE THREAD

**Issue by PatrickGaul** [Sandmann79/xbmc#671]
**Addon used**
- [ ] Amazon VOD (plugin.video.amazon-test)
- [x] Browser Launcher (plugin.program.browser.launcher)

- Addon version: 

**Account type**
- [ ] primevideo.com
- [ ] amazon.(com/co.uk/de/jp)

**System Setup (please provide the following information):**
- Hardware:  Lenovo IdeaPad Y500
- OS version: Windows 10
- Kodi version number: 20.0

**Upload Logs**

**Describe the bug**

Tried the Browser launcher today on Kodi 20.0, resulting in a "**Browser Launcher error**". I have uploaded the log to the link below.  It had been working fine on Kodi 19.4 prior to the upgrade.

[kodi.log](https://github.com/Sandmann79/xbmc/files/10679086/kodi.log)

**Comment by Sandmann79** [comment]
Has been fixed with a09da37e118a4c2b60349215a1a7ca9c66fad54b

**Comment by sergeolkhovik** [comment]
Hi! Having similar problem (but don't know if there's any similar open bug as any search on "browser launcher PythonToCppException plugin.program.browser.launcher" shows full list of bugs including VOD etc).
Installed plugin, configured it to use Chrome by default, added some URL (title, url, stop: no, kiosk: yes), then I try to open that item from the list and get plugin settings dialog.
In the log I see:

```
2024-04-02 23:20:03.173 T:15093    info <general>: CAddonMgr::FindAddon: plugin.program.browser.launcher v1.2.6+matrix.1 installed
...
2024-04-02 23:20:08.001 T:3164     info <general>: Loading skin file: DialogAddonSettings.xml, load type: KEEP_IN_MEMORY
2024-04-02 23:20:24.739 T:3164     info <general>: Loading skin file: MyPrograms.xml, load type: KEEP_IN_MEMORY
2024-04-02 23:20:25.020 T:15199 warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
2024-04-02 23:20:26.905 T:3164     info <general>: Loading skin file: DialogKeyboard.xml, load type: KEEP_IN_MEMORY
2024-04-02 23:20:26.959 T:3766     info <general>: JSONRPC Server: Disconnection detected
2024-04-02 23:21:11.222 T:3164     info <general>: Loading skin file: DialogSelect.xml, load type: KEEP_IN_MEMORY
2024-04-02 23:21:17.615 T:15201   error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'UnicodeEncodeError'>
                                                   Error Contents: 'ascii' codec can't encode characters in position 75-82: ordinal not in range(128)
                                                   Traceback (most recent call last):
                                                     File "/home/kodi/.kodi/addons/plugin.program.browser.launcher/default.py", line 278, in <module>
                                                       addSite()
                                                     File "/home/kodi/.kodi/addons/plugin.program.browser.launcher/default.py", line 125, in addSite
                                                       fh = open(os.path.join(siteFolder, getFileName(title)+".link"), 'w')
                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                   UnicodeEncodeError: 'ascii' codec can't encode characters in position 75-82: ordinal not in range(128)
                                                   -->End of Python script error report<--
                                                   
2024-04-02 23:21:43.166 T:15418 warning <general>: Setting most video properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagVideo.
```

So I have Ubuntu Mate 23.10 with Kodi v20 (upgraded from v19 recently, had the same problem):

```
% dpkg -l kodi\*|grep ii
ii  kodi                              2:20.2+dfsg-4 amd64        Open Source Home Theatre (executable binaries)
ii  kodi-bin                          2:20.2+dfsg-4 amd64        Open Source Home Theatre (architecture-dependent files)
ii  kodi-data                         2:20.2+dfsg-4 all          Open Source Home Theatre (arch-independent data package)
ii  kodi-inputstream-adaptive         20.3.11+ds-1  amd64        Adaptive inputstream addon for Kodi
ii  kodi-inputstream-ffmpegdirect     20.5.0+ds1-2  amd64        FFmpegDirect inputstream addon for Kodi
ii  kodi-inputstream-rtmp             20.3.0+ds1-1  amd64        Kodi input stream addon for RTMP
ii  kodi-peripheral-joystick          20.1.3+ds-1   amd64        Kodi Joystick Library
ii  kodi-pvr-iptvsimple               20.6.1+ds-1   amd64        IPTV Simple Client Kodi PVR Addon
ii  kodi-repository-kodi              2:20.2+dfsg-4 all          Open Source Home Theatre (official addons repository feed)
ii  kodi-visualization-spectrum:amd64 20.2.0+ds1-1  amd64        Spectrum visualizer addon for Kodi
% lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 23.10
Release:        23.10
Codename:       mantic
```

What could be wrong? Also I found that I cannot enter non-English title (i.e. in cyrillic like 'Київстар ТВ') and probably that error in log is related this problem.
Yes, correct, just tested again - when I try to open saved item (and then I see a settings dialog) I don't see any changes in the log file, but new message if I try to add new item with cyrillic name.
OK, but why the plugin doesn't open the saved URL and there's no log/debug info at all?
